File: programme/swap-two-numbers-without-using-third-variable/swap-without-temp.java
// Write a program to swap two numbers without using a third variable
import java.util.Scanner;

public class Swap {
    private static void swap(int a, int b) {
        System.out.println("\nNumbers before swap : " + a + " and " + b);
        a = a + b;
        b = a - b;
        a = a - b;
        System.out.println("\nNumbers after swap  : " + a + " and " + b);
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter a number : ");
        int a = sc.nextInt();
        System.out.print("Enter another number : ");
        int b = sc.nextInt();

        swap(a, b);
        sc.close();
    }
}

File: programme/swap-two-numbers-without-using-third-variable/swap_without_temp.go
// Write a program to swap two numbers without using third variable
package main

import "fmt"

func swap(a, b int) (int, int) {
	return b, a
}

func main() {
	var a, b int

	fmt.Print("Enter the first number a  : ")
	fmt.Scan(&a)

	fmt.Print("Enter the second number b : ")
	fmt.Scan(&b)

	fmt.Printf("\nInput  : a=%d b=%d", a, b)

	a, b = swap(a, b)
	fmt.Printf("\nOutput : a=%d b=%d\n", a, b)
}
